tests/unit/src/test/scala/tests/cli/BaseCliSuite.scala

```diff
@@ -46,4 +49,132 @@ abstract class BaseCliSuite extends FunSuite with DiffAssertions {
       onStdout(stdout)
     }
   }
+
+  def assertOneSameOrPrintExpected(obtained: String, expected: List[String], title: String = "")(
+      implicit source: Position
+  ): Boolean = {
+    try assertAllDiff(obtained, expected, title)
+    catch {
+      case ex: Exception =>
+        obtained.linesIterator.toList match {
+          case head +: tail =>
+            println("    \"\"\"|" + head)
+            tail.foreach(line => println("       |" + line))
+          case head +: Nil =>
+            println(head)
+          case Nil =>
+            println("obtained is empty")
+        }
+        throw ex
+    }
+  }
+
+  // scala.meta.testkit.DiffAssertions
+  // Cannot access these because they are private
+  import collection.JavaConverters._
+  import org.scalatest.exceptions.{StackDepthException, TestFailedException}
+
+  def splitIntoLines(string: String): Seq[String] =
+    string.trim.replace("\r\n", "\n").split("\n")
+
+  def compareContents(original: String, revised: String): String =
+    compareContents(splitIntoLines(original), splitIntoLines(revised))
+
+  def compareContents(original: Seq[String], revised: Seq[String]): String = {
+    val diff = difflib.DiffUtils.diff(original.asJava, revised.asJava)
+    if (diff.getDeltas.isEmpty) ""
+    else
+      difflib.DiffUtils
+        .generateUnifiedDiff(
+          "obtained",
+          "expected",
+          original.asJava,
+          diff,
+          1
+        )
+        .asScala
+        .mkString("\n")
+  }
+
+  case class DiffFailure(
+      title: String,
+      expected: String,
+      obtained: String,
+      diff: String,
+      source: Position
+  ) extends TestFailedException(
+        { _: StackDepthException =>
+          Some(title + "\n" + error2message(obtained, expected))
+        },
+        None,
+        source
+      )
+
+  def header[T](t: T): String = {
+    val line = s"=" * (t.toString.length + 3)
+    s"$line\n=> $t\n$line"
+  }
+
+  def stripTrailingWhitespace(str: String): String = str.replaceAll(" \n", "∙\n")
+
+  private def error2message(obtained: String, expected: String): String = {
+    val sb = new StringBuilder
+    if (obtained.length < 1000) {
+      sb.append(
+        s"""#${header("Obtained")}
+           #${stripTrailingWhitespace(obtained)}
+           #
+           #""".stripMargin('#')
+      )
+    }
+    sb.append(
+      s"""#${header("Diff")}
+         #${stripTrailingWhitespace(compareContents(obtained, expected))}""".stripMargin('#')
+    )
+    sb.toString()
+  }
+
+  def assertAllDiff(obtained: String, expected: List[String], title: String = "")(
+      implicit source: Position
+  ): Boolean = {
+    val theSame = expected.filter(p => compareContents(obtained, p).isEmpty)
+    val result = expected.foldLeft("")((acc, p) => acc + compareContents(obtained, p))
+    if (theSame.nonEmpty) true
+    else {
+      throw DiffFailure(title, expected.mkString(" OR \n"), obtained, result, source)
+    }
+  }
+
+  def checkCliMulti(
```

Comment:
Did you try using the `checkCli` method? It supports processing multiple files.  I'm curious to understand why this method is needed

Reply:
The problem is the ordering in which the files are processed.  Assume files A and B. If A gets processed before B, then A gets a count of 1 and B a 2. If the order is reversed then the B gets count 1 and A gets 2. The order in which the files are processed in my local machine and in the CI/CD VM were different. So  I crated a `checkCliMulti` method, which checks that at least one of the conditions I stipulate is true. I suspect that the transformation tests you have don't have this issue. Maybe I don't need this?

Reply:
I have just realized I don't need this test. I was using it when I had the pre and post process that were removed. So I am removing this.